### PR TITLE
[SPARK-24926][CORE] Ensure numCores is used consistently in all netty configurations

### PR DESCRIPTION
**Pull request by NiharS**
## What changes were proposed in this pull request?

Netty could just ignore user-provided configurations. In particular, spark.driver.cores would be ignored when considering the number of cores available to netty (which would usually just default to Runtime.availableProcessors() ). In transport configurations, the number of threads are based directly on how many cores the system believes it has available, and in yarn cluster mode this would generally overshoot the user-preferred value.

## How was this patch tested?

As this is mostly a configuration change, tests were done manually by adding spark-submit confs and verifying the number of threads started by netty was what was expected.

Passes scalastyle checks from dev/run-tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by squito** [comment]
OK to test

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -2668,17 +2668,24 @@ object SparkContext extends Logging {
...
+  private[spark] def numDriverCores(master: String, conf: SparkConf = null): Int = {
     def convertToInt(threads: String): Int = {
       if (threads == "*") Runtime.getRuntime.availableProcessors() else threads.toInt
     }
     master match {
       case "local" => 1
       case SparkMasterRegex.LOCAL_N_REGEX(threads) => convertToInt(threads)
       case SparkMasterRegex.LOCAL_N_FAILURES_REGEX(threads, _) => convertToInt(threads)
-      case _ => 0 // driver is not used for execution
+      case "yarn" =>
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: I think you can just write this as `case "yarn" if (conf ...) =>`; not sure if it gets too long. Or else I'd move the `else` up a line to fix that style below.

**Comment by NiharS** [comment]
Thanks for the review and feedback! I made the changes, except for the moving the if clause to the same line as "yarn", unfortunately that does make the line 104 characters long.

**Review by srowen — COMMENTED** [review]
There are other usages of `fromSparkConf` that seem to default to numCores = 0 in order to let Netty decide. Should these also be limited? See ExternalShuffleService for instance.

**Pull request closed without merging**